beam-cpp-example/pipeline.py

```diff
@@ -0,0 +1,155 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import argparse
+import logging
+import os
+import sys
+
+
+def run(argv):
+  # Import here to avoid __main__ session pickling issues.
+  import io
+  import itertools
+  import matplotlib.pyplot as plt
+  import collatz
+
+  import apache_beam as beam
+  from apache_beam.io import restriction_trackers
+  from apache_beam.options.pipeline_options import PipelineOptions
+
+  class RangeSdf(beam.DoFn, beam.RestrictionProvider):
+    """An SDF producing all the integers in the input range.
+
+    This is preferable to beam.Create(range(...)) as it produces the integers
+    dynamically rather than materializing them up front.  It is an SDF to do
+    so with perfect dynamic sharding.
+    """
+    def initial_restriction(self, desired_range):
+      start, stop = desired_range
+      return restriction_trackers.OffsetRange(start, stop)
+
+    def restriction_size(self, _, restriction):
+      return restriction.size()
+
+    def create_tracker(self, restriction):
+      return restriction_trackers.OffsetRestrictionTracker(restriction)
+
+    def process(self, _, active_range=beam.DoFn.RestrictionParam()):
+      for i in itertools.count(active_range.current_restriction().start):
+        if active_range.try_claim(i):
+          yield i
+        else:
+          break
+
+  class GenerateIntegers(beam.PTransform):
+    def __init__(self, start, stop):
+      self._start = start
+      self._stop = stop
+
+    def expand(self, p):
+      return (
+          p
+          | beam.Create([(self._start, self._stop + 1)])
+          | beam.ParDo(RangeSdf()))
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--start', dest='start', type=int, default=1)
+  parser.add_argument('--stop', dest='stop', type=int, default=10000)
+  parser.add_argument('--output', default='./out.png')
+
+  known_args, pipeline_args = parser.parse_known_args(argv)
+  # Store this as a local to avoid capturing the full known_args.
+  output_path = known_args.output
+
+  with beam.Pipeline(options=PipelineOptions(pipeline_args)) as p:
+
+    # Generate the integers from start to stop (inclusive).
+    integers = p | GenerateIntegers(known_args.start, known_args.stop)
+
+    # Run them through our C++ function, filtering bad records.
+    # Requires apache beam 2.34 or later.
+    stopping_times, bad_values = (
+        integers
+        | beam.Map(collatz.total_stopping_time).with_exception_handling(
```

Comment:
This not demonstrating using an existing cpp library. (Unlike https://github.com/apache/beam/tree/master/examples/java/src/main/java/org/apache/beam/examples/subprocess ) 

This is fine I think, but I wanted to point out the diffenrece.